src/PagesController.php

```diff
@@ -0,0 +1,276 @@
+<?php
+/**
+ * Pages Controller
+ *
+ * @author    Pronamic <info@pronamic.eu>
+ * @copyright 2005-2023 Pronamic
+ * @license   GPL-3.0-or-later
+ * @package   Pronamic\WordPress\Pay
+ */
+
+namespace Pronamic\WordPress\Pay;
+
+/**
+ * Pages Controller class
+ */
+class PagesController {
+	/**
+	 * Setup.
+	 */
+	public function setup() {
+		\add_action( 'init', [ $this, 'init' ] );
+
+		\add_action( 'admin_init', [ $this, 'admin_init' ] );
+	}
+
+	/**
+	 * Initialize.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		// Pages.
+		$pages = $this->get_pages();
+
+		foreach ( $pages as $id => $label ) {
+			\register_setting(
+				'pronamic_pay',
+				$id,
+				[
+					'type'              => 'integer',
+					'sanitize_callback' => [ $this, 'sanitize_published_post_id' ],
```

Comment:
```
Fatal error: Uncaught Error: call_user_func_array(): Argument #1 ($callback) must be a valid callback, class Pronamic\WordPress\Pay\PagesController does not have a method "sanitize_published_post_id"
```

The method `sanitize_published_post_id( $value )` is defined in the `Settings` class:
https://github.com/pronamic/wp-pay-core/blob/b943516046b593cc1efb0383694d19c26428b9ed/src/Settings.php#L140-L152

Reply:
Changed in https://github.com/pronamic/wp-pay-core/commit/b19332388b7131a4547df57dbbfe39f7a4042362.